### PR TITLE
Call mPulse bundle using UserContext

### LINKED ISSUE #1630: Add cookie check to mPulse beacon
**Is your feature request related to a problem? Please describe.**
If a user has disabled performance cookies, the mPulse beacon will still be shown, therefore not respecting a user's privacy settings.

**Describe the solution you'd like**
Check that the user has enabled cookies (exact perferences/type of cookie to check TBD) before rendering the mPulse beacon.

Check the Dynatrace Beacon in RN to see which cookie was required to be set before loading the beacon.

**Describe alternatives you've considered**
N/A

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

**Additional context**

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/simorgh/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by dr3**
Resolves #1630

**Overall change:** Call mPulse bundle using UserContext

Behaviour <a href="https://media.giphy.com/media/dTuk8wcL9XryNrKtjZ/source.mp4" target="_blank"> 
https://media.giphy.com/media/dTuk8wcL9XryNrKtjZ/source.mp4
</a>

**Code changes:**

- Removes mPulse snippet from head
- Calls it if/when user has accepted personalisation cookies
- Creates UserContext which holds info about bbc cookie

How to test: 
Enable mpulse here https://github.com/bbc/simorgh/blob/mPulseCookieLock-Context/src/app/lib/config/toggles/index.js#L10

Use a real api key here https://github.com/bbc/simorgh/blob/mPulseCookieLock-Context/envConfig/local.env#L6

Mpulse should only fire when 
`the toggle is enabled`
`the api key is set to any value`
`on client`
`user has accepted personalisation cookie`

The beacons shouldnt fire untill the user has accepted the cookie banner. If they have and they refresh, they should fire instantly. It should fire a beacon for every page view and in-simorgh navigation (article to article)

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/MPulseBeacon/boomr.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
+/* eslint-disable */
+
+const boomr = (apiKey) => {
+  console.log('BOOMR INIT');
```
Comment (chris-hinds) [inline src/app/containers/MPulseBeacon/boomr.js]: do we need this?

File: src/app/containers/MPulseBeacon/index.jsx [hunk 2]
```diff
@@ -1,13 +1,25 @@
-import React, { useContext } from 'react';
+import { useEffect, useContext } from 'react';
 import { DialContext } from '../../contexts/DialContext';
```
Comment (chris-hinds) [inline src/app/containers/MPulseBeacon/index.jsx]: can we remove the use of Dials and use a toggle instead please.

File: src/app/contexts/UserContext/index.jsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React, { useState } from 'react';
+import { node } from 'prop-types';
+import { getCookiePolicy, personalisationEnabled } from './cookies';
+
+export const UserContext = React.createContext({});
+
+export const UserContextProvider = ({ children }) => {
+  const [cookiePolicy, setCookiePolicy] = useState(getCookiePolicy());
+
+  const updateCookiePolicy = () => setCookiePolicy(getCookiePolicy());
```
Comment (chris-hinds) [inline src/app/contexts/UserContext/index.jsx]: I think we should move this to an action reducer patter n down the road, as we should not be directly updating the state object. Not a problem right now though.

**Review by chris-hinds — DISMISSED** [review]
Just a couple comments

**Review by chris-hinds — APPROVED** [review]

**Comment by dr3** [comment]
```
codeclimate/diff-coverage — 25% (80% threshold)
codeclimate/total-coverage — 90%
```

Low coverage due to the removal of files. 90% shows no lost coverage

**Comment by PriyaKR** [comment]
How is the mpulse beacon revoked when the user is on page who has initially accepted the cookies and then later changes the cookies preference?

**Review by thekp — APPROVED** [review]
rip `dials`

**Comment by PriyaKR** [comment]
The mpulse beacon gets fired acccording to the functional cookies value instead of the performance cookies value(mentioned over here https://github.com/bbc/simorgh/issues/1630#issuecomment-502007130).

**Review by chris-hinds — APPROVED** [review]

**Review by thekp — APPROVED** [review]

**Comment by PriyaKR** [comment]
Looks good to me.

When the performance cookie is enabled the events are fired to the mpulse beacon and when it's not enabled mpulse events are not fired.

Also if the user changes their cookie preference outside of Simorgh then they would need to refresh the Simorgh pages to see the changes - looks good.

**Pull request merged**